components/forms/input-fields/input-field.scss

```diff
@@ -2,167 +2,152 @@
+      &:focus {
+        outline: 4px solid rgba(0, 153, 218, 0.5);
+        border: 1px solid $clear-blue;
       }
-
-      & ~ label {
-        color: $gray-7;
+      &:focus + label {
+        top: 10px;
+        font-size: 10px;
+      }
+      &:not(:placeholder-shown) + label {
+        top: 10px;
+        font-size: 10px;
       }
-      background: $gray-3;
     }
-
-    &--readonly {
-      color: $black;
+    &__label {
+      position: absolute;
+      top: 18px;
+      left: 20px;
+      transition: 0.5s;
       pointer-events: none;
-      & ~ label {
-        color: $black;
+      color: $gray-8;
+      font-size: 16px;
+    }
+    &__info-message {
+      font-size: 12px;
+      line-height: 18px;
+      color: $gray-9;
+      margin-left: 3px;
+      margin-top: 6px;
+    }
+    &--small {
+      .msds-input__text-input {
+        padding-left: 12px;
+        height: 36px;
+        font-size: 12px;
+        line-height: 18px;
+        padding-top: 12px;
+        top: 5px;
+        & + label {
+          left: 13px;
+          font-size: 12px;
+          top: 10px;
+        }
+        &:focus + label {
+          left: 13px;
+          top: 1px;
+          font-size: 11px;
+        }
+        &:not(:placeholder-shown) + label {
+          font-size: 11px;
+          top: 1px;
+        }
       }
     }
-
     &--success {
-      border: 1px solid $confirm-green;
-      & ~ p {
-        color: $confirm-green;
-      }
-
-      & ~ p:before,
-      & ~ p:after {
+      .msds-input__icon-wrapper:before,
+      .msds-input__icon-wrapper:after {
         content: '';
         position: absolute;
         right: 15px;
+        top: calc(50% - 6px);
+        height: 13px;
         width: 2px;
         background-color: $confirm-green;
       }
-
-      & ~ p:before {
-        top: calc(50% - 16px);
+      .msds-input__icon-wrapper:before {
+        top: calc(50% - 7px);
         height: 13px;
         right: 15px;
         transform: rotate(45deg);
       }
-
-      & ~ p:after {
-        top: calc(50% - 12px);
+      .msds-input__icon-wrapper:after {
+        top: calc(50% - 3px);
         height: 7px;
-        right: 21px;
+        right: 22px;
         transform: rotate(-45deg);
       }
-
-      &:focus {
-        outline: 4px solid rgba(60, 192, 3, 0.5);
+      .msds-input__text-input {
         border: 1px solid $confirm-green;
+        &:focus {
+          outline: 4px solid rgba(60, 192, 3, 0.5);
+          border: 1px solid $confirm-green;
+        }
+      }
+      .msds-input__info-message {
+        color: $confirm-green;
       }
     }
-
     &--error {
-      border: 1px solid $danger-red-1;
-      & ~ p {
-        color: $danger-red-1;
-      }
-
-      & ~ p:before,
-      & ~ p:after {
+      .msds-input__icon-wrapper:before,
+      .msds-input__icon-wrapper:after {
         content: '';
         position: absolute;
         right: 15px;
-        top: calc(50% - 17px);
+        top: calc(50% - 6px);
```

Comment:
Wasn't it this calc position that was causing the bug with the text and icon?

Reply:
I can't hardcode it unfortunately as it is tied to the hight of the input field which can vary as we have 2 different sizes for it. else I have to create a separate markup and CSS for the small input class